lib/s3-s3.js

```diff
@@ -40,9 +46,25 @@ var s3 = function (primaryS3, secondaryS3) {
           setupOn(realRequest, 'extractError', onFunc);
           setupOn(realRequest, 'extractData', onFunc);
           setupOn(realRequest, 'success', onFunc);
-          setupOn(realRequest, 'error', onFunc);
           setupOn(realRequest, 'complete', onFunc);
 
+          // error is a special on() call which we use to see if this is a call that needs to
+          // go to the secondary
+          realRequest.on('error', function(error) {
+            if (usingPrimary && error &&
+                error.hasOwnProperty('code') &&
+              (error.code === 400 || error.code >= 500)) {
```

Comment:
There are tons of conditions that S3 lists as 400, all sorts of weird things.  So my guess is that S3 could use that generic condition for them when there's issues.  That one though... even testing won't tell us for sure or not.  Just my hunch.